prov/rxm/src/rxm_ep.c

```diff
@@ -887,10 +1029,23 @@ rxm_ep_send_common(struct rxm_ep *rxm_ep, const struct iovec *iov, void **desc,
 	assert(count <= rxm_ep->rxm_info->tx_attr->iov_limit);
 
 	ret = ofi_cmap_get_handle(rxm_ep->util_ep.cmap, dest_addr, &handle);
-	if (OFI_UNLIKELY(ret))
+	if (OFI_UNLIKELY(ret)) {
+		if (OFI_UNLIKELY(ret != -FI_EAGAIN))
+			return ret;
+		fastlock_acquire(&rxm_ep->util_ep.cmap->lock);
```

Comment:
This reuses the lock for connection management which is independent from postponed queue processing. Perhaps we should use a separate lock?

Reply:
Thanks. Yes, it is an issues here. The postponed queue wasn't protected

I'm favor of idea to use lock for connection managment, but with some changes. Please, see updated code.